libs/commons/Concurrency.mli

```diff
@@ -0,0 +1 @@
+val with_lock : (unit -> 'a) -> Mutex.t -> 'a
```

Comment:
im amazed this doesn't already exist in our library somewhere

Reply:
Maybe it does. But Common.with_open_xxx didn't exist so I expected this one neither.